src/uproot/models/RNTuple.py

```diff
@@ -538,13 +552,15 @@ def arrays(
             [c.num_entries for c in clusters[start_cluster_idx:stop_cluster_idx]]
         )
 
-        form = self.to_akform().select_columns(filter_names)
+        form = self.to_akform().select_columns(
+            filter_names, prune_unions_and_records=False
+        )
         # only read columns mentioned in the awkward form
         target_cols = []
         container_dict = {}
         _recursive_find(form, target_cols)
         for key in target_cols:
-            if "column" in key:
+            if "column" in key and "union" not in key:
```

Comment:
Does `key` have the word `"union"` in it by construction? Is it because the form analyzed by

https://github.com/scikit-hep/uproot5/blob/25fbb3a85aa87b6d817c40a362ef581c693c5cea/src/uproot/models/RNTuple.py#L608-L617

is a Form that you made and deliberately added the string `"union"`?

I'm asking because this is not something that can be generically expected of given data, only if you built it that way in this code.

Reply:
Since for variants now it initially uses nested `IndexedOptionForm(UnionForm(...))`, I added a `"-union"` suffix to differentiate the key of the inner form. Do you think this could cause issues?

https://github.com/scikit-hep/uproot5/blob/25fbb3a85aa87b6d817c40a362ef581c693c5cea/src/uproot/models/RNTuple.py#L401-L404